Week 5/Solution.java

```diff
@@ -0,0 +1,86 @@
+import java.io.*;
+import java.util.*;
+import org.junit.Test;
+import static org.junit.Assert.assertEquals;
+public class Solution {
+	//Node structure considered for this problem
+	class Node{
+		char value;
+		Node next;
+		Node(char value, Node next){
+			this.value = value;
+			this.next = next;
+		}
+	}
+    public boolean isPalindrome(Node head) {
+    	//input validation
+    	if(head == null) throw new NullPointerException();
+    	//getting the size of the linked list
+    	int ln = getLength(head);
+    	//Structure to store the first half of the list
+    	Stack<Character> first_half = new Stack<Character>();
+    	//Size of the first half. If the linked list's size is odd,the middle is n/2 rounded 
+    	//to the upper closest integer
+    	int md = (int)Math.ceil((double)ln/2);
+    	//storing the first half
+    	for(int i = 0; i<md; i++){
+    		first_half.push(head.value);
+    		head = head.next;
+    	}
+    	//If the size is odd, it doesn't consider the middle element, so
+    	//it is popped from the stack
+    	if(ln%2!=0) first_half.pop();
+    	//Compares the upper element of the stack with the next 
+    	//value until the stack is empty or they are not the same
+    	while(!first_half.isEmpty()){
+    		char curr = first_half.pop();
+    		if(curr!=head.value) return false;
+    		head = head.next;
+    	}
+    	//if we reach this point the list is a palindrome so we return true
+    	return true;
+    }
+    public int getLength(Node head){
```

Comment:
I like how you separated this into a different method! and good job with the really detailed comments :)

Reply:
thanks!